src/listener/server-profile-templates.js

```diff
@@ -203,20 +216,19 @@ export default class ServerProfileTemplateListener extends Listener {
   }
 
   FixCompliance(msg) {
-    if(this.client.connection.isReadOnly()) {
+    let host = msg.host;
+    if(this.client.isReadOnly()) {
       return this.transform.text(msg, "I'm afraid you'll have to set readOnly mode to false in your config file first if you want to do that...");
     }
 
     let dialog = this.switchBoard.startDialog(msg);
 
-    let nameAndHyperlink = getDeviceNameAndHyperLink("/rest/server-profile-templates/" + msg.templateId);
-    let templateName = nameAndHyperlink.deviceName;
-    let templateHyperlink = nameAndHyperlink.hyperlink;
-    this.transform.text(msg, msg.message.user.name + " I am going to fix the compliance issues for the profile template " + this.transform.hyperlink(templateHyperlink, templateName) + ".  Are you sure you want to do this? (@" + this.robot.name + " yes/@" + this.robot.name + " no)");
+    let nameAndHyperlink = getDeviceNameAndHyperLink(host + "/rest/server-profile-templates/" + msg.templateId);
+    this.transform.text(msg, msg.message.user.name + " I am going to fix the compliance issues for the profile template " + this.transform.hyperlink(nameAndHyperlink.hyperlink, nameAndHyperlink.deviceName) + ".  Are you sure you want to do this? (@" + this.robot.name + " yes/@" + this.robot.name + " no)");
```

Comment:
Hmm...good question. Not sure what to do about this now.  We may have to think about changing lots of our messaging now.